src/main/java/com/divudi/bean/common/OpdPreSettleController.java

```diff
@@ -291,29 +291,126 @@ private void saveSettlingBatchBill() {
         getSaleBill().setBillClassType(BillClassType.BilledBill);
         getSaleBill().setBillType(BillType.OpdBill);
         getSaleBill().setBillTypeAtomic(BillTypeAtomic.OPD_BATCH_BILL_PAYMENT_COLLECTION_AT_CASHIER);
-
         getSaleBill().setDepartment(getSessionController().getLoggedUser().getDepartment());
         getSaleBill().setInstitution(getSessionController().getLoggedUser().getDepartment().getInstitution());
-
         getBillBean().setPaymentMethodData(getSaleBill(), getSaleBill().getPaymentMethod(), paymentMethodData);
-
+        getBillBean().setBills(billsOfBatchBillPre);
```

Comment:
here we have to assign the bills newly created with the batch bill, not those of preBill